src/ForkCMS/Utility/Thumbnails.php

```diff
@@ -0,0 +1,160 @@
+<?php
+
+namespace ForkCMS\Utility;
+
+use Imagine\Gd\Imagine;
+use Imagine\Image\Box;
+use Imagine\Image\ImageInterface;
+use Imagine\Image\Point;
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Component\Finder\Finder;
+
+class Thumbnails
+{
+    /** @var Finder */
+    private $finder;
+
+    /** @var Filesystem */
+    private $filesystem;
+
+    /** @var Imagine */
+    private $imagine;
+
+    /** @var string */
+    private $sitePath;
+
+    public function __construct(string $sitePath)
+    {
+        $this->sitePath = realpath($sitePath);
+        $this->finder = new Finder();
+        $this->filesystem = new Filesystem();
+        $this->imagine = new Imagine();
+    }
+
+    /**
+     * Delete thumbnails based on the folders in the path
+     *
+     * @param string $inPath The path wherein the thumbnail-folders exist.
+     * @param string|null $filename The filename to be deleted.
+     */
+    public function delete(string $inPath, ?string $filename): void
+    {
+        // if there is no image filename provided we can't do anything
+        if ($filename === null || $filename === '') {
+            return;
+        }
+
+        foreach ($this->finder->directories()->in($inPath) as $directory) {
+            $fileName = $directory->getRealPath() . '/' . $filename;
+            if (is_file($fileName)) {
+                $this->filesystem->remove($fileName);
+            }
+        }
+    }
+
+    /**
+     * Generate thumbnails based on the folders in the path
+     * Use
+     *  - 128x128 as foldername to generate an image where the width will be
+     *      128px and the height will be 128px
+     *  - 128x as foldername to generate an image where the width will be
+     *      128px, the height will be calculated based on the aspect ratio.
+     *  - x128 as foldername to generate an image where the height will be
+     *      128px, the width will be calculated based on the aspect ratio.
+     *
+     * @param string $inPath The path wherein the thumbnail-folders will be stored.
+     * @param string $forSourceFile The location of the source file.
+     */
+    public function generate(string $inPath, string $forSourceFile): void
+    {
+        // get folder listing
+        $folders = $this->getFolders($inPath);
```

Comment:
why don't you loop over the array that the method returns directly?

Reply:
Fixed